libs/application/templates/general-fishing-license/src/lib/dataSchema.ts

```diff
@@ -0,0 +1,52 @@
+import * as kennitala from 'kennitala'
+import * as z from 'zod'
+import { error } from './messages/error'
+
+export const GeneralFishingLicenseSchema = z.object({
+  approveExternalData: z.boolean().refine((v) => v),
+  externalData: z.object({
+    nationalRegistry: z.object({
+      data: z.object({
+        address: z.object({
+          city: z.string(),
+          code: z.string(),
+          postalCode: z.string().refine((x) => +x >= 100 && +x <= 999, {
+            params: error.invalidValue,
+          }),
+          streetAddress: z.string(),
+        }),
+        age: z.number(),
+        citizenship: z.object({
+          code: z.string(),
+          name: z.string(),
+        }),
+        fullName: z.string(),
+        legalResidence: z.string(),
+        nationalId: z.string(),
+      }),
+      date: z.string(),
+      status: z.enum(['success', 'failure']),
+    }),
+    userProfile: z.object({
+      data: z.object({
+        email: z.string(),
+        mobilePhoneNumber: z.string(),
+      }),
+    }),
+  }),
```

Comment:
The dataSchema only applies to `answers`

Reply:
Truthfully, me just being lazy and saving myself a step. I removed it and will add it once I add the data providers. 😄